elastic/store: use wait instead of get for barrier

d4l3k opened the pull request:
Summary: We call `.get` in the elastic store barrier operation but we don't need the result. This switches it to use `.wait` instead which eliminates one network round trip as `get` internally does a wait first.

Test Plan:

CI + existing tests -- no behavior change

Differential Revision: D59396199


cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @chauhang

kurman reviewed (approved):
good catch, stamping to unblock - @d4l3k can you fix the store unit tests mocking before landing?

d4l3k committed:
elastic/store: use wait instead of get for barrier (#130148)

Summary:
We call `.get` in the elastic store barrier operation but we don't need the result. This switches it to use `.wait` instead which eliminates one network round trip as `get` internally does a wait first.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/130148

Test Plan:
CI + existing tests -- no behavior change

cc mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab chauhang

Reviewed By: kurman

Differential Revision: D59396199

Pulled By: d4l3k
